src/main.rs

```diff
@@ -631,23 +626,12 @@ fn truncate(s: &str, n: usize) -> String {
 
     if s.len() <= n { return String::from(s) }
 
-    let mut truncated = String::new();
-    for grapheme in UnicodeSegmentation::graphemes(s, true).take(n - tail.len()) {
-        truncated.push_str(grapheme)
-    }
-    truncated.push_str(tail);
-    truncated
+    UnicodeSegmentation::graphemes(s, true).take(n - tail.len()).collect::<String>() + tail
```

Comment:
Awesome, I don't think I realized you could collect into a String.  I also think I wrote this before I figured out you could use + for string contatenation.